samples/scripts/syncAnnotationsScript.py

```diff
@@ -0,0 +1,261 @@
+
+5. Now options are presented to the user, they can either upload the annotations to the source directory or they can create a folder in their User/Public
+directory with a custom name and clone the datasets to add annotations to.
+
+6. Once the decisions to either create a User/Public folder or upload to the dataset source locations is made the system will then upload the annotations.
+
+7. This is where the 'default_label' option is used.  If the the folder under Annotations doesn't match the 'default_label' when uploading to the source images
+it will create a clone with the name {Dataset} - {label}.  This allows multiple folders to be used to upload different annotations to the system.
+"""
+
+
+apiURL = "localhost"
+port = 8010
+baseGirderId = ""  # Set by the second argument to the script
+baseGirderType = "folder"  # folder | collection | user
+keyFolderName = "ANNOTATIONS"  # The folder name under which to look for Annotations
+keyFileName = "annotations-"  # a key for replacement on the annotation-file to sync it with the dataset
+default_label = "unlabeled"  # when uploading this will be the folder that will replace the existing annotations.
+# all other labels will create a clone with the name of the dataset and the label type.
+
+
+# Login to the girder client, interactive means it will prompt for username and password
+def login():
+    gc = girder_client.GirderClient(apiURL, port=port, apiRoot="girder/api/v1")
+    gc.authenticate(interactive=True)
+    return gc
+
+
+# Simple search within folder to get annotation files and their paths
+def get_annotations(folder):
+    checkFolder = folder
+    foundfiles = []
+    for root, dirs, files in os.walk(checkFolder):
+        for file in files:
+            if file.endswith(".csv"):
+                # base name exists we can get a list of files
+                foundfiles.append(os.path.join(root, file))
+    return foundfiles
+
+
+# returns the girder base path for the indexed folder
+def getBasePath(gc: girder_client.GirderClient, folder: str, type: str):
+    return gc.sendRestRequest("GET", f"resource/{folder}/path?type={type}")
+
+
+# attempts to find the filename dataset and the 'Video {filename}'
+def find_dataset(gc: girder_client.GirderClient, filename: str, baseGirderPath: str):
+    basename = os.path.basename(filename)
+    # first we need to remove the ANNOTATIONS/{labeled}/annotations from the filename
+    print(filename)
+    start_index = filename.find(keyFolderName)
+    end_index = filename.find(keyFileName) + len(keyFileName)
+    removed_label = filename[start_index:end_index].split(os.path.sep)[1]
+    removed_data = filename[end_index:]
+    check_path = removed_data.replace(".csv", "")
+    check_path = check_path.replace(basename, f"Video {basename}")
+    # Check for a Video
+    found = gc.sendRestRequest(
+        "GET",
+        "resource/lookup",
+        parameters={"path": f'/{baseGirderPath}/{check_path.replace(".csv", ".mp4")}'},
+    )
```

Comment:
I personally think this is too specific of a use case and we should consider different paths, different naming conventions etc. However if this is just a template for people to be able to change to their own circumstances I am fine with that.

Reply:
There is the sibling script to this for uploading videos/annotations which can be used a more basic template without having to parse this particular folder structure.